src/config.py

```diff
@@ -0,0 +1,32 @@
+from dependency_injector import containers, providers
+
+from src.algorithms.dbscan import DBSCAN
+from src.algorithms.kmeans import KMeans
+from src.algorithms.spectral_clustering import SpectralClustering
+from src.file_handlers.json_handler import JsonHandler
+from src.file_handlers.numpy_handler import NumpyHandler
+
+
+class ConfigContainer(containers.DeclarativeContainer):
+    """Container for managing configuration settings."""
+
+    config = providers.Configuration()
+
+    get_algorithm = providers.Selector(
+        selector=config.clustering.type,
+        kmeans=providers.Factory(KMeans.create, config.clustering.kmeans),
```

Comment:
you should write explicitly what parameters are expected in class initialization